### PR TITLE
CIV-13278 Judgment Online Service

### PR DESCRIPTION
**Pull request by ShwetaTandel-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-13278

### Change description ###

JO changes

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/DefaultJudgementSpecHandler.java [hunk 1]
```diff
@@ -417,28 +419,23 @@ private CallbackResponse generateClaimForm(CallbackParams callbackParams) {
...
+        if (featureToggleService.isJudgmentOnlineLive() && JudgmentsOnlineHelper.isNonDivergent(caseData)) {
             nextState = CaseState.All_FINAL_ORDERS_ISSUED.name();
             caseDataBuilder.businessProcess(BusinessProcess.ready(DEFAULT_JUDGEMENT_NON_DIVERGENT_SPEC));
         } else {
             nextState = CaseState.PROCEEDS_IN_HERITAGE_SYSTEM.name();
             caseDataBuilder.businessProcess(BusinessProcess.ready(DEFAULT_JUDGEMENT_SPEC));
         }
+        if (featureToggleService.isJudgmentOnlineLive()) {
+            caseDataBuilder.activeJudgment(djOnlineMapper.addUpdateActiveJudgment(caseData));
+            caseDataBuilder.joIsLiveJudgmentExists(YesOrNo.YES);
```
Comment (hmcts-version1-ignacio) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/DefaultJudgementSpecHandler.java:432]: Extra line?

Comment (ShwetaTandel-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/DefaultJudgementSpecHandler.java:432]: Where .. Line number pls?

**Review by hmcts-version1-ignacio — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/helpers/judgmentsonline/JudgmentsOnlineHelper.java [hunk 2]
```diff
@@ -67,4 +72,69 @@ public static List<String> validateMidCallbackData(CaseData caseData) {
...
+            || MultiPartyScenario.isTwoVOne(caseData)
+            || (ofNullable(caseData.getRespondent2()).isPresent()
+            && ofNullable(caseData.getDefendantDetailsSpec()).isPresent()
+            && ofNullable(caseData.getDefendantDetailsSpec().getValue()).isPresent()
+            && caseData.getDefendantDetailsSpec().getValue().getLabel().startsWith("Both"));
+    }
+
+    public static boolean isNonDivergentForJBA(CaseData caseData) {
+        return  MultiPartyScenario.isOneVOne(caseData)
+            || MultiPartyScenario.isTwoVOne(caseData)
```
Comment (sjamihmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/helpers/judgmentsonline/JudgmentsOnlineHelper.java:86]: Currently we are routing the claims to Non-Divergent flow of JBA to keep them Online(ALL_FINAL_ORDERS_ISSUED) state only if they are LRvLiPOnevOne, All other claims would go offline. Should we check for 2v1 as well?

Comment (ShwetaTandel-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/helpers/judgmentsonline/JudgmentsOnlineHelper.java:86]: No that is diffrent requirment related to state. This method only is used for marking Judgment active or not.

Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/helpers/judgmentsonline/JudgmentsOnlineHelper.java:86]: ok. Fine..

**Review by sjamihmcts — APPROVED** [review]

**Review by Omaira-Melo-Hmcts — APPROVED** [review]

**Review by Azam-Hmcts — APPROVED** [review]

**Review by Omaira-Melo-Hmcts — APPROVED** [review]

**Pull request merged**